### PR TITLE
✨ Update ListFiles API to return error

### PR DESCRIPTION
**Pull request by azeemshaikh38**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
ListFiles API now returns error along with filename slice.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No.

### CODE REVIEW

File: checks/checkforcontent.go [hunk 1]
```diff
@@ -68,20 +68,24 @@ func CheckFilesContent(checkName, shellPathFnPattern string,
...
 		b, err := isMatchingPath(shellPathFnPattern, filepath, caseSensitive)
 		if err != nil {
-			return false
+			return false, err
 		}
-		return b
+		return b, nil
 	}
 	res := true
-	for _, file := range c.RepoClient.ListFiles(predicate) {
```
Comment (laurentsimon) [inline checks/checkforcontent.go:84]: I think `CheckFilesContent()` is no longer used. I have not removed it yet as part of post-migration cleanup. Feel free to remove it.

File: checks/checkforfile.go [hunk 2]
```diff
@@ -22,7 +22,11 @@ import (
 // for the occurrence.
 func CheckIfFileExists(checkName string, c *checker.CheckRequest, onFile func(name string,
```
Comment (laurentsimon) [inline checks/checkforfile.go:23]: CheckIfFileExists is no longer used. I think you can remove it as part of this PR.

**Review by laurentsimon — APPROVED** [review]

File: checks/checkforcontent.go (hunk 1 above)
Comment (azeemshaikh38) [reply checks/checkforcontent.go:84]: Done.

File: checks/checkforfile.go (hunk 2 above)
Comment (azeemshaikh38) [reply checks/checkforfile.go:23]: Done.

**Pull request merged**